### PR TITLE
extended https://github.com/OpenSC/OpenSC/pull/754

### PR DESCRIPTION
**Pull request by frankmorgner**

### CODE REVIEW

File: src/libopensc/pkcs15-gemsafeGPK.c [hunk 1]
```diff
@@ -191,7 +191,7 @@ static int sc_pkcs15emu_gemsafeGPK_init(sc_pkcs15_card_t *p15card)
 	};
 
 	const prdata prkeys[] = {
-		{ "1", "AUTH key", 1024, USAGE_AUT, "I0009",
+		{ "01", "AUTH key", 1024, USAGE_AUT, "I0009",
 		  0x00, "1", 0},
```
Comment (dengert) [inline src/libopensc/pkcs15-gemsafeGPK.c]: 195 should read 
0x00, "01", 0),

its the auth_id

File: src/libopensc/pkcs15-gemsafeGPK.c [hunk 2]
```diff
@@ -283,7 +283,7 @@ static int sc_pkcs15emu_gemsafeGPK_init(sc_pkcs15_card_t *p15card)
 		}
 
 		kinfo[num_keyinfo].fileid = i;
-		sc_pkcs15_format_id("NONE", &kinfo[num_keyinfo].id); 
+		sc_pkcs15_format_id("FF", &kinfo[num_keyinfo].id);
```
Comment (dengert) [inline src/libopensc/pkcs15-gemsafeGPK.c]: Why did you pick "FF". I think the 0.15.0 current code would would not change the id, which may be uninitialized 
A beter choice might be "", that should cause sc_hex_to_bin to set id->len=0;
But  Later in line 416, the cert_info.id is copied to the kinfo[n].id so it get overwritten anyway. 

(I wrote this code in 2005, but don't have any of these cards. I doubt the the code is even used by anyone.)

**Comment by frankmorgner** [comment]
fixed

**Comment by dengert** [comment]
Your updated changes look good to me, I will merge into my repro and run a few tests, then update #754 later today.

**Comment by dengert** [comment]
OK, merged your changes in to my #754  waiting for test to run.

**Comment by mouse07410** [comment]
1. How do I download it for my testing? 
2. In @frankmorgner 's additions I saw nothing that was PIV-related - it seemed all for aligning other drivers with the new `sc_hex_to_bin()` requirements. So perhaps not relevant for our problem?

**Comment by dengert** [comment]
I have merged @frankmorgner 's changes into #754, 
so look at #754 again. The piv was already there.

**Pull request merged**